### DISCUSSION TITLE
0.9.2: how to ignore `Unschedulable`?

### DISCUSSION THREAD

**Discussion by mfn** [abahmed/kwatch#310] [Q&A]
Ever since I upgraded to 0.9.0 (but now on 0.9.2) `CronJob` which can't be immediately scheduled are reported as "Errors: Reason `Unschedulable`".

I've a staging system where I also use kwatch, where often it takes multiple tries to schedule CronJob, therefore I want to ignore those errors.

For similar reasons I already have an existing config, here's my current (pre 0.9.0) `ConfigMap`:
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: kwatch
  namespace: kwatch
data:
  config.yaml: |
    alert:
      slack:
        webhook: …
        title: k8s-staging
    maxRecentLogLines: 10
    reasons:
      - '!CreateContainerConfigError'
```
Now I want to add `Unschedulable` to be ignored too, but nothing I tried worked:
- a separate line
  ```yaml
      reasons:
        - '!CreateContainerConfigError'
        - '!Unschedulable'
  ```
- added to the existing one
  ```yaml
      reasons:
        - '!CreateContainerConfigError,Unschedulable'
  ```

So far no luck: those errors are still coming.

I wrote:
> Ever since I upgraded to 0.9.0

Basically, I'm pretty sure I was never notified of `Unschedulable` before (using 0.8.5).

Any idea what I'm doing wrong?

**Comment by mfn** [comment]
Tested also with 0.9.3, but no luck.

Does no one have a clue how to ignore `CreateContainerConfigError` & `Unschedulable` with the YAML syntax?

**Comment by abahmed** [comment]
Hi @mfn — we’ve been working on significant enhancements and improvements to kwatch, including better filtering and suppression behavior, but those changes have not been released yet.

In the upcoming version, the recommended approach will be:

```yaml
silences:
  - reasons:
      - Unschedulable
  - eventMessages:
      - "failed to sync configmap cache"
```

This suppresses `Unschedulable` and the specific transient cache-sync error while keeping other `CreateContainerConfigError` incidents visible.
